server/src/services/DiscordClient.js

```diff
@@ -127,13 +126,22 @@ module.exports = class DiscordClient {
           `User ${user.username} (${user.id}) in allowed users list, skipping guild and role check.`,
         )
       } else {
+        const guildsFull = user.guilds
         for (let i = 0; i < this.strategy.blockedGuilds.length; i += 1) {
           const guildId = this.strategy.blockedGuilds[i]
           if (guilds.includes(guildId)) {
-            perms.blocked = !!guildsFull.find((x) => x.id === guildId)
-            return perms
+            perms.blocked = true
+            const currentGuildName = guildsFull.find(
+              (x) => x.id === guildId,
+            ).name
+            perms.blockedGuildNames = perms.blockedGuildNames
+              ? perms.blockedGuildNames.add(currentGuildName)
+              : new Set([currentGuildName])
           }
         }
+        if (perms.blocked === true) {
+          return perms
```

Comment:
```suggestion
          perms.blockedGuildNames = Array.from(perms.blockedGuildNames)
          return perms
```
Lets turn the set into an array here to be consistent with the other perms. Then we can do just do normal array operations in the other places you're using `Array.from()`.

Reply:
For reference you're actually using a Set for the webhook & scanner perms. https://github.com/WatWowMap/ReactMap/pull/729/files#diff-fcad83d8a0f164f00fdbb33b303964ac83270bada9a2e90e1078aec2b46f13a5R109-R113

That that said you are using spread operators instead of `Array.from()` in the rest of the codebase so I have switched to that.

Reply:
Sorry for the confusion, I am using sets within this function but I turn them all into arrays before returning https://github.com/WatWowMap/ReactMap/blob/d812c48b255a1d83102c26e458edf00a0ceedf35/server/src/services/DiscordClient.js#L197, is what I was referencing.

Reply:
ew. I didn't see that.

Reply:
Resolved properly now